File: 1752-check-if-array-is-sorted-and-rotated/1752-check-if-array-is-sorted-and-rotated.java
class Solution {
    public boolean check(int[] nums) {
        int size=nums.length,x=0;
        int[] b=Arrays.copyOf(nums,size);
        Arrays.sort(b);
        for(int i=1;i<size;i++)
        {
            if(nums[i]<nums[i-1])
            {
                x=i;
                break;
            }
        }
        for(int i=0;i<size;i++)
        {
            if(b[i]!=nums[(i+x)%size])
                return false;
        }
        return true;
    }
}

File: 1752-check-if-array-is-sorted-and-rotated/1752-check-if-array-is-sorted-and-rotated.cpp
class Solution {
public:
    bool check(vector<int>& nums) {
        int pivot=0;
        for(int i=0;i<nums.size()-1;i++){
            if(nums[i]>nums[i+1])
                pivot++;
            if(pivot>1)
                return false;
        }
        return (nums[0]<nums[nums.size()-1] && pivot>0)?false:true;
    }
};